Utilities/Files.cs

```diff
@@ -23,49 +25,66 @@ public class Files
         public static string Read(string name)
         {
             string result = null;
+            int attempts = 10;
+            int ioDelayMs = 25;
             if (name != null)
             {
-                try
+                while (attempts > 0)
                 {
-                    result = File.ReadAllText(name, Encoding.UTF8);
-                }
-                catch (ArgumentException ex)
-                {
-                    Logging.Error("Failed to read from " + name, ex);
-                }
-                catch (PathTooLongException ex)
-                {
-                    Logging.Error("Path " + name + " too long", ex);
-                }
-                catch (DirectoryNotFoundException ex)
-                {
-                    Logging.Error("Directory for " + name + " not found", ex);
-                }
-                catch (FileNotFoundException ex)
-                {
-                    if (!ignoreMissing)
+                    try
                     {
-                        Logging.Error("File " + name + " not found", ex);
+                        result = File.ReadAllText(name, Encoding.UTF8);
                     }
-                }
-                catch (IOException ex)
-                {
-                    if (!ignoreMissing)
+                    catch (ArgumentException ex)
                     {
-                        Logging.Error("IO exception for " + name, ex);
+                        Logging.Error("Failed to read from " + name, ex);
```

Comment:
Wont these loop infinitely, since attempts is not decremented or zeroed?

Reply:
We break out of the loop, regardless of whether `attempts` decrements to zero, using `break;` at line 83.

Reply:
No, that would change the logic.

I think the structure is very confusing and fragile. 
1. Only `catch (IOException ex) when ((ex.HResult & 0x0000FFFF) == 32) // Sharing violation` with `ignoreMissing == false` has a path that decrements `attempts`. 
2. All other catch statements and the happy path lead to a `break`.

I think the control logic needs a bit of a re-think here.